cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_impl_dpc.cpp

```diff
@@ -0,0 +1,1230 @@
+        }
+        if constexpr (check_mask_flag(bs_list::sum | bs_list::mean | sum2cent_based_stat, List)) {
+            lsum_ptr[id] = mrgsum;
+        }
+        if constexpr (check_mask_flag(bs_list::sum2 | bs_list::sorm, List)) {
+            lsum2_ptr[id] = mrgsum2;
+        }
+        if constexpr (check_mask_flag(bs_list::mean | sum2cent_based_stat, List)) {
+            lrc_ptr[id] += rcnt;
+        }
+        if constexpr (check_mask_flag(sum2cent_based_stat, List)) {
+            lsum2cent_ptr[id] = mrgsum2cent;
+        }
+        if constexpr (check_mask_flag(bs_list::mean | sum2cent_based_stat, List)) {
+            lmean_ptr[id] = mrgmean;
+        }
+    }
+
+    for (std::int64_t stride = sycl::min(local_size, block_count) / 2; stride > 0; stride /= 2) {
+        item.barrier(sycl::access::fence_space::local_space);
+
+        if (stride > id) {
+            std::int64_t offset = id + stride;
+
+            Float min = Float(0);
+            if constexpr (check_mask_flag(bs_list::min, List)) {
+                min = lmin_ptr[offset];
+            }
+            Float max = Float(0);
+            if constexpr (check_mask_flag(bs_list::max, List)) {
+                max = lmax_ptr[offset];
+            }
+            Float sum = Float(0);
+            if constexpr (check_mask_flag(bs_list::sum | bs_list::mean | sum2cent_based_stat,
+                                          List)) {
+                sum = lsum_ptr[offset];
+            }
+            Float sum2 = Float(0);
+            if constexpr (check_mask_flag(bs_list::sum2 | bs_list::sorm, List)) {
+                sum2 = lsum2_ptr[offset];
+            }
+            std::int64_t rcnt = 1;
+            if constexpr (check_mask_flag(bs_list::mean | sum2cent_based_stat, List)) {
+                rcnt = lrc_ptr[offset];
+            }
+            Float sum2cent = Float(0);
+            if constexpr (check_mask_flag(sum2cent_based_stat, List)) {
+                sum2cent = lsum2cent_ptr[offset];
+            }
+            Float mean = Float(0);
+            if constexpr (check_mask_flag(bs_list::mean | sum2cent_based_stat, List)) {
+                mean = lmean_ptr[offset];
+            }
+
+            Float sum_n1n2 = mrgvectors + static_cast<Float>(rcnt);
+            Float mul_n1n2 = mrgvectors * static_cast<Float>(rcnt);
+            Float delta_scale = mul_n1n2 / sum_n1n2;
+            Float mean_scale = Float(1) / sum_n1n2;
+            Float delta = mean - mrgmean;
+
+            mrgmin = sycl::fmin(min, mrgmin);
+            mrgmax = sycl::fmax(max, mrgmax);
+            mrgsum += sum;
+            mrgsum2 += sum2;
+            mrgsum2cent = mrgsum2cent + sum2cent + delta * delta * delta_scale;
+            mrgmean = (mrgmean * mrgvectors + mean * static_cast<Float>(rcnt)) * mean_scale;
+            mrgvectors = sum_n1n2;
+
+            // item 0 collects all results in private vars
+            // but all others need to store it
+            if (0 < id) {
```

Comment:
Maybe it is just a matter of taste but please use variable first in comparisons. It will accentuate reader on this variable.

Reply:
The idea is set const first to prevent errors like if(id = 0), yes, it isn't relevant for this particular case due to id is const, but it is better to use it in one manner for all such cases.